app/src/main/java/com/google/samples/apps/sunflower/compose/garden/GardenScreen.kt

```diff
@@ -0,0 +1,265 @@
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.compose.garden
+
+import androidx.activity.compose.ReportDrawnWhen
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.rememberLazyGridState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Card
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import com.google.samples.apps.sunflower.viewmodels.GardenPlantingListViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.bumptech.glide.integration.compose.ExperimentalGlideComposeApi
+import com.google.accompanist.themeadapter.material.MdcTheme
+import com.google.samples.apps.sunflower.R
+import com.google.samples.apps.sunflower.compose.card
+import com.google.samples.apps.sunflower.compose.utils.SunflowerImage
+import com.google.samples.apps.sunflower.data.GardenPlanting
+import com.google.samples.apps.sunflower.data.Plant
+import com.google.samples.apps.sunflower.data.PlantAndGardenPlantings
+import com.google.samples.apps.sunflower.viewmodels.PlantAndGardenPlantingsViewModel
+import java.util.*
+
+@Composable
+fun GardenScreen(
+    viewModel: GardenPlantingListViewModel = viewModel(),
+    onAddPlantClick: () -> Unit,
+    onPlantClick: (PlantAndGardenPlantings) -> Unit
+) {
+    val gardenPlants by viewModel.plantAndGardenPlantings.collectAsState(initial = emptyList())
+    GardenScreen(
+        gardenPlants = gardenPlants,
+        onAddPlantClick = onAddPlantClick,
+        onPlantClick = onPlantClick
+    )
+}
+
+@Composable
+private fun GardenScreen(
+    gardenPlants: List<PlantAndGardenPlantings>?,
```

Comment:
this list won't be null anymore, right? because you pass `initial = emptyList()`